tests/metrics/github/test_security.py

```diff
@@ -3,85 +3,239 @@
-            ],
+            "createdAt": "2022-02-10T01:36:54Z",
+            "fixedAt": None,
+            "dismissedAt": None,
+        },
+        {
+            "createdAt": "2022-10-18T17:20:30Z",
+            "fixedAt": "2022-10-24T14:27:29Z",
+            "dismissedAt": None,
+        },
+        {
+            "createdAt": "2022-10-18T21:08:22Z",
+            "fixedAt": "2022-11-09T13:01:04Z",
+            "dismissedAt": None,
+        },
+        {
+            "createdAt": "2022-11-01T17:52:53Z",
+            "fixedAt": "2022-11-07T15:14:37Z",
+            "dismissedAt": None,
+        },
+        {
+            "createdAt": "2023-08-30T04:44:56Z",
+            "fixedAt": None,
+            "dismissedAt": "2023-09-04T15:07:44Z",
+        },
+        {
+            "createdAt": "2023-10-03T02:46:00Z",
+            "fixedAt": None,
+            "dismissedAt": None,
+        },
+        {
+            "createdAt": "2023-10-26T15:02:17Z",
+            "fixedAt": None,
+            "dismissedAt": None,
         },
     ]
     return response
 
 
-def test_security_number_of_alerts_today():
-    today = date(2023, 11, 28)
+def test_get_repos():
+    security.query_repos = fake_repos
+    security.query_vulnerabilities = fake_vulnerabilities
+    fake_client = lambda: None
+    fake_client.org = "test-org"
+
+    result = security.get_repos(fake_client)
+
+    assert len(result) == 2
+    assert result[0].name == "opencodelists"
+    assert len(result[0].vulnerabilities) == 7
+    assert result[1].name == "job-server"
+    assert len(result[1].vulnerabilities) == 7
+
+
+def test_repo_earliest_date():
+    vulnerabilities = [
+        security.Vulnerability(date(2023, 10, 26), None, None),
+        security.Vulnerability(date(2023, 10, 29), None, None),
+    ]
+    repo = security.Repo("test", "test-org", vulnerabilities)
+
+    assert repo.earliest_date() == date(2023, 10, 26)
+
+
+def test_repo_latest_date():
+    vulnerabilities = [
+        security.Vulnerability(date(2023, 10, 26), None, None),
+        security.Vulnerability(date(2023, 10, 29), None, None),
+    ]
+    repo = security.Repo("test", "test-org", vulnerabilities)
+
+    assert repo.latest_date() == date(2023, 10, 29)
+
+
+def test_vulnerability_open_at():
+    v = security.Vulnerability(date(2023, 10, 26), None, None)
+
+    assert v.is_open_at(date(2023, 10, 29))
+
 
-    alerts = fake_vulnerabilities()[0]["vulnerabilities"]
-    result = security.parse_vulnerabilities_for_date(
-        alerts, "opencodelists", today, "test-org"
-    )
+def test_vulnerability_open_at_same_day():
+    v = security.Vulnerability(date(2023, 10, 26), None, None)
 
-    assert str(result["date"]) == "2023-11-28"
-    assert result["closed"] == 4
-    assert result["open"] == 3
+    assert v.is_open_at(date(2023, 10, 26))
+
+
+def test_vulnerability_open_at_date_in_past():
+    v = security.Vulnerability(date(2023, 10, 26), None, None)
+
+    assert v.is_open_at(date(2023, 10, 20)) is False
```

Comment:
Do you have a preference for `assert ... is False` over `assert not ...`? If so, then I'll keep it in mind 🙂

Reply:
No real preference, I think `assert not` is probably how most people would do it and more pythonic, so I've changed it to that.